### PR TITLE
#7475 Refactor deploymentUpdater

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Part of #7475 (preliminary refactor before starting)
- Renames instances of extension -> modComponent
- Renames instances of install -> activate
- Consolidates redundant abstraction for resetting the updatePromptTimestamp

## Checklist

- [x] Add tests and/or storybook stories
- [x] Designate a primary reviewer @BLoe

### CODE REVIEW

File: src/background/deploymentUpdater.ts [hunk 1]
```diff
@@ -621,45 +622,51 @@ async function activateDeploymentsInBackground({
...
-  if (manual.length > 0 || automaticError) {
+  if (deploymentsToManuallyActivate.length > 0 || autoActivationError) {
     void browser.runtime.openOptionsPage();
   }
 }
 
 /**
- * Reset the update countdown timer on startup.
+ * There is a prompt in the UI shown to the user to encourage them to manually activate and/or update deployments,
+ * controlled by updatePromptTimestamp. Set updatePromptTimestamp to null to effectively hide the prompt.
```
Comment (mnholtz) [inline src/background/deploymentUpdater.ts:656]: > Set updatePromptTimestamp to null to effectively hide the prompt.

See "Open Question" in the PR description. I'm not 100% sure if this claim is actually right? After 30 minutes of looking at the code i wasn't confident what the intended behavior of setting updatePromptTimestamp to `null` actually is.

File: src/store/settings/settingsTypes.ts [hunk 2]
```diff
@@ -87,7 +87,9 @@ export type SettingsStateV1 = SkunkworksSettingsFlags &
     nextUpdate: number | null;
 
     /**
-     * Timestamps the update modal was first shown. Used to calculate time remaining for enforceUpdateMillis
+     * Controls a prompt in the UI that prompts the user to manually activate and/or update deployed mods.
+     * This represents the timestamp the update modal was first shown. Set to null to show the modal for the first time.
+     * Used to calculate time remaining for enforceUpdateMillis.
```
Comment (mnholtz) [inline src/store/settings/settingsTypes.ts:92]: See https://github.com/pixiebrix/pixiebrix-extension/pull/8136/files#r1548324605

File: src/background/deploymentUpdater.ts [hunk 3]
```diff
@@ -88,146 +88,153 @@ const locateAllForService = locator.locateAllForService.bind(locator);
...
   optionsState: ModComponentState,
   editorState: EditorState | undefined,
-  recipeId: RegistryId,
-): Promise<{
-  options: ModComponentState;
-  editor: EditorState | undefined;
-}> {
+): Promise<{ options: ModComponentState; editor: EditorState | undefined }> {
   let options = optionsState;
   let editor = editorState;
```
Comment (grahamlangford) [inline src/background/deploymentUpdater.ts:217]: NIT: `options` and `editor` are ambiguous. 

```suggestion
  let _optionsState = optionsState;
  let _editorState = editorState;
```

File: src/background/deploymentUpdater.ts (hunk 1 above)
Comment (twschiller) [reply src/background/deploymentUpdater.ts:656]: > to effectively hide the prompt.

What does "effectively" mean here (what's the intent)? It hides the prompt if it was already showing? It resets the snooze timer?

Example call here: https://github.com/pixiebrix/pixiebrix-extension/blob/fdb8834d3f60f8dedf1c8206b26af47ab826e9bd/src/extensionConsole/pages/deployments/DeploymentsContext.tsx#L183-L183

File: src/background/deploymentUpdater.ts [hunk 4]
```diff
@@ -621,45 +622,54 @@ async function activateDeploymentsInBackground({
...
  * - We don't currently separately track timestamps for showing an update modal for deployments vs. browser extension
  * upgrades. However, in enterprise scenarios where enforceUpdateMillis is set, the IT policy is generally such
  * that IT can't reset the extension.
+ *
+ * @see DeploymentModal
  */
-async function resetUpdatePromptTimestamp() {
-  // There could be a race here, but unlikely because this is run on startup
-  console.debug("Resetting updatePromptTimestamp");
+async function hideUpdatePromptUntilNextAvailableUpdate() {
```
Comment (mnholtz) [inline src/background/deploymentUpdater.ts:666]: Just pointing out this change here `hideUpdatePromptUntilNextAvailableUpdate`

File: src/background/deploymentUpdater.ts [hunk 5]
```diff
@@ -621,45 +622,54 @@ async function activateDeploymentsInBackground({
...
   }
 }
 
 /**
- * Reset the update countdown timer on startup.
+ * There is a prompt in the UI shown to the user to encourage them to manually activate and/or update deployments,
+ * partially controlled by updatePromptTimestamp. Set updatePromptTimestamp to null to hide the modal until
+ * deployment updates are next available.
  *
  * - If there was a Browser Extension update, it would have been applied
```
Comment (mnholtz) [inline src/background/deploymentUpdater.ts:659]: BTW I have no idea what either of these comments mean exactly. I tentatively left them for now

**Review by BLoe — APPROVED** [review]

**Pull request merged**